src/Compilers/Core/Portable/Symbols/ISymbolExtensions_PerformIVTCheck.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Collections;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static partial class ISymbolExtensions
+    {
+        /// <summary>
+        /// Given that an assembly with identity assemblyGrantingAccessIdentity granted access to assemblyWantingAccess,
+        /// check the public keys to ensure the internals-visible-to check should succeed. This is used by both the
+        /// C# and VB implementations as a helper to implement `bool IAssemblySymbol.GivesAccessTo(IAssemblySymbol toAssembly)`.
+        /// </summary>
+        internal static IVTConclusion PerformIVTCheck(
+            this AssemblyIdentity assemblyGrantingAccessIdentity,
+            ImmutableArray<byte> assemblyWantingAccessKey,
+            ImmutableArray<byte> grantedToPublicKey)
+        {
+            // This gets a bit complicated. Let's break it down.
+            //
+            // First off, let's assume that the "other" assembly is Smith.DLL, that the "this"
+            // assembly is "Jones.DLL", and that Smith has named Jones as a friend (that is a precondition
+            // to calling this method). Whether we allow Jones to see internals of Smith depends on these four factors:
+            //
+            // q1) Is Smith strong-named?
```

Comment:
Nit: Could I suggest "Granting.dll" and "Wanting.dll" to align on terminology? (no "this"/"other", "Smith"/"Jones")

Reply:
I tried doing that and now the comment is readable!

---
In reply to: [187128624](https://github.com/dotnet/roslyn/pull/26727#discussion_r187128624) [](ancestors = 187128624)